baremetalpxeserver/baremetalpxeserver/pxeserveragent.py

```
@@ -193,12 +202,20 @@ def _get_mac_address(ifname: str) -> str:
 
     @staticmethod
     def _get_ip_address(ifname: str) -> str:
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        return socket.inet_ntoa(fcntl.ioctl(
-            s.fileno(),
-            0x8915,  # SIOCGIFADDR
-            struct.pack('256s', ifname[:15].encode())
-        )[20:24])
+        """获取接口的第一个可用 IP（优先 IPv4，若无则取 IPv6 全局地址）。"""
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)